components/style/values/generics/position.rs

```diff
@@ -104,23 +104,23 @@ impl<L: HasViewportPercentage> HasViewportPercentage for PositionValue<L> {
 impl<L: Parse> PositionValue<L> {
     /// Internal parsing function which (after parsing) checks the keyword with the
     /// given function.
-    pub fn parse_internal<F>(context: &ParserContext, input: &mut Parser,
-                             mut is_allowed_keyword: F) -> Result<PositionValue<L>, ()>
+    pub fn parse_internal<'i, 't, F>(context: &ParserContext, input: &mut Parser<'i, 't>,
+                             mut is_allowed_keyword: F) -> Result<PositionValue<L>, ParseError<'i>>
         where F: FnMut(Keyword) -> bool
     {
         let (mut pos, mut keyword) = (None, None);
         for _ in 0..2 {
             if let Ok(l) = input.try(|i| L::parse(context, i)) {
                 if pos.is_some() {
-                    return Err(())
+                    return Err(StyleParseError::UnspecifiedError.into())
```

Comment:
Yes. It's a nice source of easy issues and a lot of effort for this patch in particular.